design/metallb-operator.md

```diff
@@ -0,0 +1,62 @@
+# MetalLB Operator
+
+## Summary
+
+MetalLB currently includes [instructions and resources for deploying
+MetalLB](https://metallb.universe.tf/installation/) using a few methods:
+directly with manifests, customized manifests with kustomize, or via helm.
+Another approach for managing the deployment of software on Kubernetes is via
+an [operator](https://operatorhub.io/what-is-an-operator).
+
+This document proposes creating an operator for MetalLB in a new git repository
+under the metallb github organization and publishing it on
+[OperatorHub.io](https://operatorhub.io/).
+
+## Motivation
+
+For any Kubernetes environment that primarily uses operators to manage add-ons
+to the cluster, it would be beneficial to have the option of using an operator
+to manage MetalLB, as well.
+
+In [another proposal](https://github.com/metallb/metallb/pull/833), we have
+discussed creating a CRD based configuration interface for MetalLB. It seems
+that settling on the ideal data model may require some more time and
+experimentation. We can make use of this separate operator repo as a place to
+try out a CRD interface for MetalLB which writes out a ConfigMap and doesn't
+have to disrupt to core MetalLB project.
```

Comment:
+1 on this. The only caveat is having a good plan if we consider moving the "experimental" CRD over to metallb itself.
I would aim at having the same configuration CRD, and in case moving it as it is to metallb if / when it gets accepted.
This would avoid the need of supporting both apis imo

Reply:
Yeah, that would be the hope ... that we end up with something that's working well enough that we consider migrating it over to MetalLB.  if we need to take a fresh look at it and a v2 of the CRDs, we'd hopefully implement an automated migration if technically feasible.  It's hard to predict, but nobody wants to make our own job harder than necessary, and that includes inflicting pain on our users. :)

Reply:
that is actually the current plan since the functionality scope of this config CRD will be limited we will be taking few  parts that are more or less agreed on from #850 and use it here